### ISSUE TITLE
[Feature] XIRR for Fixed rate Investments

### ISSUE THREAD

**Issue by ryannamani** [actualbudget/actual#1754]
### Verified feature request does not already exist?

- [X] I have searched and found no existing issue

### 💻

- [ ] Would you like to implement this feature?

### Pitch: what problem are you trying to solve?

Ability to track & calculate the Individual & Overall XIRR of multiple Investments, which are at some Fixed rate of interest & also post transactions if any partial redumption are paid-out

Please find the below example data

![image](https://github.com/actualbudget/actual/assets/6128116/df09334e-8065-4153-9ed2-ecc3bc05dd45)

### Describe your ideal solution to this problem

Basically an Off-Budget account with Investment tracking functionality

- Add a scheduled transaction from Bank Account to Off-Budget "Investment Account"
- Off-Budget accounts should be able to project the Future Cashflow based on the inputs (Intrest rate, period, date & amount)
- Should be able to post an redumption transaction from Off-Budget "Investment Account" to Bank Account there by re-calculating the XIRR of that particular Investment & Overall XIRR of Investment Account

**Comment by Shazib** [comment]
I have thought about implementing something like this a number of times.

My idea was a bit different, to have a load of off-budget accounts that i could map to various funds/stocks and then hooking into the yahoo API or something so it could do a balancing transaction once a day or something getting the current value.

In theory this daily automated 'transaction' could just be anything, it could run once a month with the scheduled fixed interest rate etc.

**Comment by Shazib** [comment]
It would be really powerful to be able to leverage the existing schedule functionality to essentially run custom scripts with the API, or just add more features there such as settings interest % etc.

Very much "power user" features though
